### PR TITLE
add auto port detection as in jupyter notebook

### PR DESCRIPTION
**Pull request by katsar0v**
Fixes https://github.com/voila-dashboards/voila/issues/392 :)

### CODE REVIEW

**Comment by jtpio** [comment]
Thanks @katsar0v for looking at this.

File: voila/app.py [hunk 1]
```diff
@@ -488,9 +494,41 @@ def stop(self):
...
+                    continue
+                else:
+                    raise
+            else:
+                self.port = port
+                success = True
+                break
+
+        if not success:
+            self.log.critical(_('ERROR: the notebook server could not be started because '
```
Comment (jtpio) [inline voila/app.py]: We probably want to say "voila" here.

Comment (katsar0v) [reply voila/app.py]: I made a new commit :)

File: voila/app.py [hunk 2]
```diff
@@ -488,9 +494,41 @@ def stop(self):
...
+                    self.log.warning(_("Permission to listen on port %i denied") % port)
+                    continue
+                else:
+                    raise
+            else:
+                self.port = port
+                success = True
+                break
+
+        if not success:
```
Comment (maartenbreddels) [inline voila/app.py]: could be a single else. e.g.
```
for port in ...
   ...
else:
  ...
```

**Pull request merged**